fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QueryBuilderUtil.java

```diff
@@ -149,6 +155,20 @@ private static ZonedDateTime addOneMinuteOrSecond(ZonedDateTime dateTime) {
         }
         return dateTime;
     }
+    private static ZonedDateTime minusOneMicroSecond(ZonedDateTime dateTime) {
+        return dateTime.minusNanos(1000);
+    }
```

Comment:
this goes away if we can either normalize the end times to always be inclusive or exclusive OR make schema edits to know the difference.

Reply:
on our call today, we decided to attempt the latter.